basic-examples/bind-constants/workflow.yaml

```diff
@@ -9,35 +9,28 @@ input:
           type:
             type_id: ref
             id: RepeatedValues
+            namespace: $.steps.foreach_loop.execute.inputs.items
         iterations:
           type:
             type_id: list
             items:
               id: SubRootObject
               type_id: ref
-    RepeatedValues:
-      id: RepeatedValues
-      properties:
-        name:
-          type:
-            type_id: string
-        ratio:
-          type:
-            type_id: float
-    SubRootObject:
-      id: SubRootObject
-      properties:
-        loop_id:
-          type:
-            type_id: integer            
+              namespace: $.steps.foreach_loop.execute.inputs.items
             
 steps:
   foreach_loop:
+    # The plugin will be called once for each entry in $.input.iterations.
     kind: foreach
+    # It will be passed an object which contains two fields, one containing 
+    # the appropriate entry from the iterations list, and one containing 
+    # the repeated input data.

```

Comment:
Since the list of "two fields" contains only two items, there's no call for a comma in line 26.

Reply:
sounds good